atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -1438,6 +1443,190 @@ public void checkImmutableTsLockAfterReadsForConservativeIfFlagIsSet() {
         timelockService.unlock(ImmutableSet.of(res.getLock()));
     }
 
+    @Test
+    public void getOrphanedSweepSentinelDoesNotThrow() {
+        Transaction t1 = txManager.createNewTransaction();
+        writeSentinelToTestTable(TEST_CELL);
+        assertThatCode(() -> t1.get(TABLE_SWEPT_CONSERVATIVE, ImmutableSet.of(TEST_CELL)))
+                .doesNotThrowAnyException();
+    }
+
+    @Test
+    public void getSweepSentinelUnderCommittedWriteThrowsAndCanBeRetried() {
+        Transaction t1 = txManager.createNewTransaction();
+        writeSentinelToTestTable(TEST_CELL);
+
+        Transaction t2 = txManager.createNewTransaction();
+        t2.put(TABLE_SWEPT_CONSERVATIVE, ImmutableMap.of(TEST_CELL, PtBytes.toBytes("blablabla")));
+        t2.commit();
+
+        assertThatThrownBy(() -> t1.get(TABLE_SWEPT_CONSERVATIVE, ImmutableSet.of(TEST_CELL)))
+                .isInstanceOf(TransactionFailedRetriableException.class)
+                .hasMessageContaining("Tried to read a value that has been deleted.");
+
+        Transaction retryTxn = txManager.createNewTransaction();
+        assertThatCode(() -> retryTxn.get(TABLE_SWEPT_CONSERVATIVE, ImmutableSet.of(TEST_CELL)))
+                .doesNotThrowAnyException();
+    }
+
+    @Test
+    public void getSweepSentinelUnderEarlyUncommittedWriteDoesNotThrow() {
```

Comment:
should swap the names of this and the following test

Reply:
ah, you're right!